cassandra-mesos-model/src/main/proto/io/mesosphere/mesos/frameworks/cassandra/model.proto

```diff
@@ -83,36 +134,89 @@ message CassandraConfigRole {
 
     // additional configuration
     // process environment
-    optional TaskEnv taskEnv = 10;
+    optional TaskEnv taskEnv = 8;
     // cassandra.yaml configuration
-    optional TaskConfig cassandraYamlConfig = 11;
+    optional TaskConfig cassandraYamlConfig = 9;
     // mesos role to be used to receive resource offers
-    optional string mesosRole = 12;
+    optional string mesosRole = 10;
 }
 
+/**
+ * One port mapping consisting of a port name and value.
+ */
 message PortMapping {
     required string name = 1;
     required int32 port = 2;
 }
 
+/**
+ * Current status of the cluster.
+ * This is a top-level object in the state hierarchy.
+ */
 message CassandraClusterState {
+    /**
+     * List of all known nodes.
+     */
     repeated CassandraNode nodes = 1;
+    /**
+     * List of all known executors.
+     */
     repeated ExecutorMetadata executorMetadata = 2;
+    /**
+     * Timestamp when the last Cassandra server run task has been submitted.
+     */
     optional int64 lastServerLaunchTimestamp = 3;
+    /**
+     * List of IP addresses of nodes that need to be replaced.
+     */
     repeated string replaceNodeIps = 4;
+    /**
+     * Number of new nodes that have to be acquired.
+     * If replaceNodeIps is not empty, new nodes will start with the -Dcassandra.replace_address=<IP>
+     * option passed to the Cassandra process.
+     */
     required int32 nodesToAcquire = 5;
     required int32 seedsToAcquire = 6;
 }
 
+/**
+ * Contains a history of the last health-checks that were received from all nodes.
+ * The implementation should not add a new entry when it is similar to the previous one.
+ * Instead it should update the timespan in the previous one.
```

Comment:
We should move:
> The implementation should not add a new entry when it is similar to the previous one. Instead it should update the timespan in the previous one.

To the "Persistent" class that implements this logic. The model object doesn't care.